File: 3663-find-the-least-frequent-digit/3663-find-the-least-frequent-digit.java
class Solution {
    public int getLeastFrequentDigit(int n) {
       List<Integer> val = new ArrayList<>();
       Map<Character,Integer> mp = new HashMap<>();
       String s = String.valueOf(n);
       for(char ch : s.toCharArray()){
        mp.put(ch,mp.getOrDefault(ch,0) + 1);
       }
       int mn = Integer.MAX_VALUE;
         for (Map.Entry<Character, Integer> entry : mp.entrySet()) {
            mn = Math.min(mn, entry.getValue());
        }

        for (Map.Entry<Character, Integer> entry : mp.entrySet()) {
            if (entry.getValue() == mn) {
                val.add(entry.getKey() - '0');
            }
        }
        return Collections.min(val);
    }
}

File: 3663-find-the-least-frequent-digit/3663-find-the-least-frequent-digit.cpp
class Solution {
public:
    int getLeastFrequentDigit(int n) {
        vector<int> val;
        map<char, int> mp;

        string s = to_string(n);

        for(char ch : s) {
            mp[ch]++;
        }

        int mn = INT_MAX;

        for(auto [k, v] : mp) {
            mn = min(mn, v);
        }

        for(auto [k, v] : mp) {
            if(v == mn) {
                val.push_back(k - '0');
            }
        }

        return *min_element(val.begin(), val.end());
    }
};